pkg/oam/workload/containerized/containerized.go

```diff
@@ -0,0 +1,256 @@
+		TypeMeta: metav1.TypeMeta{
+			Kind:       deploymentKind,
+			APIVersion: deploymentAPIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cw.GetName(),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					labelKey: string(cw.GetUID()),
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						labelKey: string(cw.GetUID()),
+					},
+				},
+			},
+		},
+	}
+	if cw.Spec.OperatingSystem != nil {
+		if d.Spec.Template.Spec.NodeSelector == nil {
+			d.Spec.Template.Spec.NodeSelector = map[string]string{}
+		}
+		d.Spec.Template.Spec.NodeSelector["beta.kubernetes.io/os"] = string(*cw.Spec.OperatingSystem)
+	}
+
+	if cw.Spec.CPUArchitecture != nil {
+		if d.Spec.Template.Spec.NodeSelector == nil {
+			d.Spec.Template.Spec.NodeSelector = map[string]string{}
+		}
+		d.Spec.Template.Spec.NodeSelector["kubernetes.io/arch"] = string(*cw.Spec.CPUArchitecture)
+	}
+
+	for _, container := range cw.Spec.Containers {
+		if container.ImagePullSecret != nil {
+			d.Spec.Template.Spec.ImagePullSecrets = append(d.Spec.Template.Spec.ImagePullSecrets, corev1.LocalObjectReference{
+				Name: *container.ImagePullSecret,
+			})
+		}
+		kubernetesContainer := corev1.Container{
+			Name:    container.Name,
+			Image:   container.Image,
+			Command: container.Command,
+			Args:    container.Arguments,
+		}
+
+		if container.Resources != nil {
+			kubernetesContainer.Resources = corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    container.Resources.CPU.Required,
+					corev1.ResourceMemory: container.Resources.Memory.Required,
+				},
+			}
+			for _, v := range container.Resources.Volumes {
+				mount := corev1.VolumeMount{
+					Name:      v.Name,
+					MountPath: v.MouthPath,
+				}
+				if v.AccessMode != nil && *v.AccessMode == oamv1alpha2.VolumeAccessModeRO {
+					mount.ReadOnly = true
+				}
+				kubernetesContainer.VolumeMounts = append(kubernetesContainer.VolumeMounts, mount)
+
+			}
+		}
+
+		for _, p := range container.Ports {
+			port := corev1.ContainerPort{
+				Name:          p.Name,
+				ContainerPort: p.Port,
+			}
+			if p.Protocol != nil {
+				port.Protocol = corev1.Protocol(*p.Protocol)
+			}
+			kubernetesContainer.Ports = append(kubernetesContainer.Ports, port)
+		}
+
+		for _, e := range container.Environment {
+			if e.Value != nil {
+				kubernetesContainer.Env = append(kubernetesContainer.Env, corev1.EnvVar{
+					Name:  e.Name,
+					Value: *e.Value,
+				})
+				continue
+			}
+			if e.FromSecret != nil {
+				kubernetesContainer.Env = append(kubernetesContainer.Env, corev1.EnvVar{
+					Name: e.Name,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							Key: e.FromSecret.Key,
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: e.FromSecret.Name,
+							},
+						},
+					},
+				})
+			}
+		}
```

Comment:
This functionality is new (i.e. doesn't currently exist in `addon-oam-kubernetes-remote`). It was scheduled to be implemented in https://github.com/crossplane/addon-oam-kubernetes-remote/pull/4 but that PR will be closed if this one is merged.